pkgs/build-support/rust/rust-utils.nix

```diff
@@ -0,0 +1,344 @@
+	cp -r target/build/* $out/rlibs # */
+      fi
+      if [ "$(ls -A target/bin)" ]; then # */
+        mkdir -p $out/bin
+        cp -P target/bin/* $out/bin # */
+      fi
+      runHook postInstall
+    '';
+
+in
+
+crate_: lib.makeOverridable ({ rust, release, verbose, crateOverrides }:
+
+let crate = crate_ // (lib.attrByPath [ crate_.crateName ] (attr: {}) crateOverrides crate_); in
+stdenv.mkDerivation rec {
+
+    inherit (crate) crateName;
+
+    src = if lib.hasAttr "src" crate then
+        crate.src
+      else
+        pkgs.fetchCrate { inherit (crate) crateName version sha256; };
+    name = "rust_${crate.crateName}-${crate.version}";
+    buildInputs = [ rust pkgs.ncurses ] ++ (crate.buildInputs or []);
+    dependencies =
+      builtins.map
+        (dep: dep.override { rust = rust; release = release; verbose = verbose; })
+        (crate.dependencies or []);
+
+    buildDependencies =
+      builtins.map
+        (dep: dep.override { rust = rust; release = release; verbose = verbose; })
+        (crate.buildDependencies or []);
+
+    completeDeps = lib.lists.unique (dependencies ++ lib.lists.concatMap (dep: dep.completeDeps) dependencies);
+    completeBuildDeps = lib.lists.unique (buildDependencies ++ lib.lists.concatMap (dep: dep.completeBuildDeps) buildDependencies);
+
+    crateFeatures = lib.concatMapStringsSep " " (f: "--cfg feature=\\\"${f}\\\"") (crate.features or []);
+
+    libName = crate.libName or crate.crateName;
+    libPath = crate.libPath or "";
+
+    metadata = builtins.substring 0 10 (builtins.hashString "sha256" (crateName + "-" + crateVersion));
+
+    crateBin =
+       builtins.foldl' (bins: bin:
+          let name =
+              lib.strings.replaceStrings ["-"] ["_"] (bin.name or crateName);
+              path = bin.path or "src/main.rs";
+          in
+          bins + (if bin == "" then "" else ",") + "${name} ${path}"
+
+       ) "" (crate.crateBin or []);
+
+    finalBins =
+       builtins.foldl' (bins: bin:
+          let name = lib.strings.replaceStrings ["-"] ["_"] (bin.name or crateName);
+              new_name = bin.name or crateName;
+          in
+          if name == new_name then bins else
+          (bins + "mv target/bin/${name} target/bin/${new_name};")
+
+       ) "" (crate.crateBin or []);
+
+    build = crate.build or "";
+    crateVersion = crate.version;
+    crateType =
+      if crate ? procMacro then "proc-macro" else
+      if crate ? plugin then "dylib" else "lib";
+    colors = crate.colors or "always";
+    buildPhase = buildCrate {
+      inherit crateName dependencies buildDependencies completeDeps completeBuildDeps
+              crateFeatures libName build release libPath crateType crateVersion
+              metadata crateBin finalBins verbose colors;
+    };
+    installPhase = installCrate crateName;
+
+}) {
+  rust = pkgs.rustc;
+  release = true;
+  verbose = true;
+  crateOverrides = import ./defaultCrateOverrides.nix { pkgs = pkgs; };
```

Comment:
Can you add a `defaultCrateOverrides` packages in `all-packages.nix` instead.
Then we can add a chapter in documentation, how to specify own overrides

```
buildRustCrate.override ({
  crateOverrides = defaultCrateOverrides // {
    my-special-crate = attr: {};
  };
})
```